tools/sqlite3_api_wrapper/test/test_sqlite3_udf_api_wrapper.cpp

```diff
@@ -0,0 +1,186 @@
+#include "catch.hpp"
+#include "sqlite3.h"
+#include <string>
+#include <thread>
+#include <string.h>
+
+#include "sqlite_db_wrapper.hpp"
+#include "sqlite_stmt_wrapper.hpp"
+
+// All UDFs are implemented in "udf_scalar_functions.hpp"
+#include "udf_scalar_functions.hpp"
+
+using namespace std;
+
+TEST_CASE("SQLite UDF wrapper: basic usage", "[sqlite3wrapper]") {
+	SQLiteDBWrapper db_w;
+
+	// open an in-memory db
+	REQUIRE(db_w.Open(":memory:"));
+
+	// create and populate table
+	REQUIRE(db_w.Execute("CREATE TABLE integers(i INTEGER)"));
+	for (int i = -5; i <= 5; ++i) {
+		// Insert values: -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5
+		REQUIRE(db_w.Execute("INSERT INTO integers VALUES (" + std::to_string(i) + ")"));
+	}
+
+	// create sqlite udf
+	REQUIRE(sqlite3_create_function(db_w.db, "multiply10", 1, 0, nullptr, &multiply10, nullptr, nullptr) == SQLITE_OK);
+	REQUIRE(db_w.Execute("SELECT multiply10(i) FROM integers"));
+	REQUIRE(db_w.CheckColumn(0, {"-50", "-40", "-30", "-20", "-10", "0", "10", "20", "30", "40", "50"}));
+}
```

Comment:
Could you add a test that results in an error (e.g. a division operator, that throws an error on zero on the rhs)?